### PR TITLE
Added MetatranscriptomicTypeV10

### PR DESCRIPTION
**Pull request by charles-cowart**

### CODE REVIEW

**Review by mmbryant23 — APPROVED** [review]

**Comment by charles-cowart** [comment]
@AmandaBirmingham @RodolfoSalido additional test is added. Thank you for your patience. Please feel free to continue reviewing!

File: metapool/__init__.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
 from .sample_sheet import (sample_sheet_to_dataframe, make_sample_sheet,
                            AmpliconSampleSheet, MetagenomicSampleSheetv90,
                            MetagenomicSampleSheetv100, AbsQuantSampleSheetv10,
-                           MetatranscriptomicSampleSheet, demux_sample_sheet,
+                           MetatranscriptomicSampleSheetv0, demux_sample_sheet,
```
Comment (RodolfoSalido) [inline metapool/__init__.py:8]: typo? Should be `MetatranscriptomicSampleSheetv10`

**Comment by charles-cowart** [comment]
@RodolfoSalido should be okay. Since we decided on this one being v10, I took our legacy MetatranscriptomicSampleSheet class and renamed it MetatranscriptomicSampleSheetv0. I thought zero’th version was somewhat appropriate.

**Comment by antgonza** [comment]
Just as a reminder, there are lost of sample sheets (not sure about versions) that we need to recover from before the SPP so my guess is that the first version of the SPP is not the actual first version ...

File: metapool/sample_sheet.py [hunk 2]
```diff
@@ -944,6 +946,54 @@ def __init__(self, path=None):
...
+            'sample sheet Sample_ID': 'Sample_ID',
+            'Sample': 'Sample_Name',
+            'Project Plate': 'Sample_Plate',
+            'Well': 'well_id_384',
+            'i7 name': 'I7_Index_ID',
+            'i7 sequence': 'index',
+            'i5 name': 'I5_Index_ID',
+            'i5 sequence': 'index2',
+            'Project Name': 'Sample_Project',
+        }
```
Comment (RodolfoSalido) [inline metapool/sample_sheet.py:994]: for absolute quant sample_sheets we included 1 dictionary key:value pair for each data_column that we expected in the samplesheet. Thus, do we need to remap `'vol_extracted_elution_ul':'vol_extracted_elution_ul'` and `'Sample RNA Concentration':'total_rna_concentration_ng_ul' `? In this case  'Sample RNA Concentration' will be the front facing column header (notebook interface) while 'total_rna_concentration_ng_ul' is the samplesheet and qiita header. Alternatively, I can name the 'Sample RNA Concentration' column with 'total_rna_concentration_ng_ul' in the notebook. Will the make_sample_sheet() function find the column defined in `data_columns` in the plate_df objects that gets passed as an argument and populate the associated data to the samplesheet?

**Review by RodolfoSalido — COMMENTED** [review]
Left a comment on sample_sheet.py to make sure we are passing along the columns ['total_rna_concentration_ng_ul','vol_extracted_elution_ul'] to the samplesheet.

**Comment by charles-cowart** [comment]
> Left a comment on sample_sheet.py to make sure we are passing along the columns ['total_rna_concentration_ng_ul','vol_extracted_elution_ul'] to the samplesheet.

Okay I'm going to circle back on this after I issue the PR resolving these columns that aren't making it from sample-sheet to prep-info file, just to make sure we implement it right here as well.

File: metapool/sample_sheet.py [hunk 3]
```diff
@@ -962,7 +1012,11 @@ def load_sample_sheet(sample_sheet_path):
...
         return sheet
 
-    sheet = MetatranscriptomicSampleSheet(sample_sheet_path)
+    sheet = MetatranscriptomicSampleSheetv10(sample_sheet_path)
+    if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
+        return sheet
+
+    sheet = MetatranscriptomicSampleSheetv0(sample_sheet_path)
     if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
         return sheet
```
Comment (AmandaBirmingham) [inline metapool/sample_sheet.py:1022]: A thought for the future: I assume that more and more defined sample sheet types/versions are in our future (given that lots and lots of not-explicitly-defined types/versions are in our past :D ).  Given this, at some point we may want to rewrite this as something like:

```
# placed somewhere easy to modify, like near the top of the module or even in a config:
ordered_sheet_factories = [AmpliconSampleSheet, MetagenomicSampleSheetv100, MetagenomicSampleSheetv90, MetatranscriptomicSampleSheetv10, MetatranscriptomicSampleSheetv0, AbsQuantSampleSheetv10]

# and then modify function to something like:
for curr_factory in ordered_sheet_factories:
    sheet = curr_factory(sample_sheet_path)
    if sheet.validate_and_scrub_sample_sheet(echo_msgs=False):
         return sheet
```

**Review by AmandaBirmingham — APPROVED** [review]
I don't know enough to have an informed opinion in this codebase, but I don't see any issues :)

File: metapool/sample_sheet.py (hunk 2 above)
Comment (charles-cowart) [reply metapool/sample_sheet.py:994]: To be honest, I don't have enough information to answer that last part. The first part is easy enough to add into the dictionary for that class, if we need it. I propose we approve this PR and test it to see if we can resolve your two questions.

File: metapool/sample_sheet.py (hunk 3 above)
Comment (charles-cowart) [reply metapool/sample_sheet.py:1022]: Will do! Ty for the suggestion!

**Comment by charles-cowart** [comment]
Hi @RodolfoSalido ! If you're okay with everything and approve this PR, I can merge it into master and we can test it against your use case. I don't expect any issues but if there is we can fix them.

**Comment by RodolfoSalido** [comment]
Sounds good to me.

**Comment by charles-cowart** [comment]
I'll resolve the issue w/lint after the meeting.

**Comment by charles-cowart** [comment]
resolved

**Comment by charles-cowart** [comment]
Thanks @RodolfoSalido ! The new Metatranscriptomic type is merged in. If you want to pull down the latest version of master and give it a try, please feel free. Or if you want to meet or zoom and test through everything together I can do that as well.

**Pull request merged**